packages/server/src/service/gm.ts

```diff
@@ -19,7 +19,32 @@ const findByType = (
            : components.find(c => c.types.includes(type))?.long_name)
 }
 
-const rawGeocode = async (query: string): Promise<google.maps.GeocoderResult | null> => {
+const rawGeocode = async (addr: AddressInputParts | string): Promise<google.maps.GeocoderResult | null> => {
+  let query: string
+  // Type assertion to ensure backward-compatibility with <StateRedirect/>
```

Comment:
To be more clear, I think you're referring to `addr` being able to be string as well.  Let's say "addr can be string to ensure ..." and put the comment in the docstring for this function.